lib/rubygems/specification.rb

```diff
@@ -193,6 +194,12 @@ class Gem::Specification < Gem::BasicSpecification
   @@spec_with_requirable_file = {}
   @@active_stub_with_requirable_file = {}
 
+  # Tracking removed method calls to warn users during build time.
+  REMOVED_METHODS = [:rubyforge_project=].freeze # :nodoc:
+  def removed_method_calls
```

Comment:
Could we use a plain Array + `uniq`? I could imagine Hiroshi making `set` a default gem, and us having trouble here because of using it... :sweat_smile: Bundler already uses `set` but `rubygems` doesn't I believe so I guess if we can avoid it we should.

Reply:
sure, np

Reply:
I have removed set and moved to array. `rubyforge_project=` should be usually called only once during gemspec evaluation.